core/src/main/java/com/linecorp/armeria/client/AbstractRuleBuilder.java

```diff
@@ -174,6 +175,18 @@ public AbstractRuleBuilder onException() {
         return onException((unused1, unused2) -> true);
     }
 
+    /**
+     * Adds {@link TimeoutException}.
+     */
+    public AbstractRuleBuilder onResponseTimeout() {
+        return onException((ctx, ex) -> {
+            if (ex instanceof UnprocessedRequestException) {
+                return ex.getCause() instanceof TimeoutException;
```

Comment:
Question) Couldn't `ctx.isTimedOut()` still be true?

```suggestion
                ex = ex.getCause();
```

Reply:
No, it isn't. It will be listed in the TODO on
https://github.com/line/armeria/issues/4876#issuecomment-1553326492